File: 2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.cpp
// Last updated: 2/28/2026, 4:25:50 PM
class Solution {
public:
   vector<int> buildArray(vector<int>& nums) 
	{
		int n = nums.size();

		for(int i=0;i<n;i++)
		{
			nums[i] = nums[i] + n*(nums[nums[i]]%n);
		}

		for(int i=0;i<n;i++)
		{
			nums[i] = nums[i]/n;
		}
		return nums;
	}
};

File: 2048-BuildArrayFromPermutation/2048-BuildArrayFromPermutation.java
// Last updated: 2/28/2026, 4:25:55 PM
class Solution {
    public int[] buildArray(int[] nums) {
        
        int[] ans = new int[nums.length];
        
        for(int i = 0 ; i < nums.length; i++){
             ans[i] = nums[nums[i]];
         } 
        
        return ans;
    }
    
}